src/client/src/reducers/wizardSelectionReducers/services/appServiceReducer.ts

```diff
@@ -0,0 +1,95 @@
+import { AZURE_TYPEKEYS } from "../../../actions/azureActions/typeKeys";
+import { messages } from "../../../selectors/wizardSelectionSelector";
+import { FormattedMessage } from "react-intl";
+import AzureActionType from "../../../actions/azureActions/azureActionType";
+import { WIZARD_INFO_TYPEKEYS } from "../../../actions/wizardInfoActions/typeKeys";
+import WizardInfoType from "../../../actions/wizardInfoActions/wizardInfoActionType";
+
+/* State Shape
+{
+    appService: {
+        siteNameAvailability: {
+          isSiteNameAvailable: boolean,
+          message: string
+        },
+        selection: ISelectedAppService | null,
+        wizardContent: {
+          serviceType: string,
+        },
+        chooseExistingRadioButtonSelected: boolean;
+    }
+}
+*/
+
+export interface IAvailability {
+  isSiteNameAvailable: boolean;
+  message: string;
+}
+
+export interface ISelectedAppService {
+  subscription: string;
+  resourceGroup: string;
+  siteName: string;
+  internalName: string;
+}
+
+interface IServiceContent {
+  serviceType: FormattedMessage.MessageDescriptor;
+}
+
+export interface IAppServiceSelection {
+  siteNameAvailability: IAvailability;
+  selection: ISelectedAppService | null;
+  wizardContent: IServiceContent;
+  chooseExistingRadioButtonSelected: boolean;
+}
+
+const initialState: IAppServiceSelection = {
+  siteNameAvailability: {
+    isSiteNameAvailable: false,
+    message: "App name unavailable"
```

Comment:
resource string?

Reply:
Okay I looked into this, this should have been a simple fix but changing this from a `string` to a `FormattedMessage.MessageDescriptor` requires lot more changes then I expected.

The action type for setting availability is coupled in `setAccountAvailability.ts` so we need to refactor all three forms:
1) changing to use intl.formatMessage, rather than this message
2) changing how we update this state
3) changing button validation 

After discussing with Tanya and Nipun, the best solution would be to just remove this message field from the reducer `siteNameAvailability`, and just keep the boolean, and then in the component render the resource string. 

Task: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/31327